docs/dqx/docs/reference/quality_rules.mdx

```diff
@@ -470,6 +470,30 @@ checks = [
 ```
 </details>
 
+### Usage examples of check functions with Map and Array columns
+
+You can apply checks to complex columns (e.g. `MapType` or `ArrayType`) by passing column expressions to the check function:
+<details style={{ backgroundColor: 'transparent', color: 'neutral' }}>
+<summary>**Checks defined on complex columns**</summary>
+```python
+from pyspark.sql.functions import col, try_element_at
+from databricks.labs.dqx.rule import DQRule
+from databricks.labs.dqx.col_functions import *
+from datetime import datetime
+
+checks = [
+    DQRule(
+        criticality="error",
+        check=is_not_null(col("map_col").getItem("map_key"))
+    ),
+    DQRule(
+        criticality="error",
+        check=is_not_null(try_element_at("array_col", 1))
+    )
+]
+```
```

Comment:
It would also be useful to provide an example that is using all items of an array instead of a single element, e.g. 
DQRule(criticality="error", check=is_not_greater_than(F.expr("array_max(transform(array_col, x -> x.key)) as max_val"), limit=10)),